File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/main/java/io/github/lishangbu/avalon/oauth2/common/web/authentication/DefaultAuthenticationSuccessHandler.java
package io.github.lishangbu.avalon.oauth2.common.web.authentication;

import io.github.lishangbu.avalon.web.util.JsonResponseWriter;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import java.util.Objects;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.Authentication;
import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
import tools.jackson.databind.json.JsonMapper;

/// 默认认证成功处理器
///
/// 在认证成功时返回一个统一的 JSON 成功响应
///
/// @author lishangbu
/// @since 2025/8/23
@Slf4j
public class DefaultAuthenticationSuccessHandler implements AuthenticationSuccessHandler {

    private final JsonMapper jsonMapper;

    public DefaultAuthenticationSuccessHandler(JsonMapper jsonMapper) {
        this.jsonMapper = Objects.requireNonNull(jsonMapper, "jsonMapper");
    }

    @Override
    public void onAuthenticationSuccess(
            HttpServletRequest request,
            HttpServletResponse response,
            Authentication authentication) {
        JsonResponseWriter.writeSuccessResponse(response, jsonMapper);
    }
}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/main/kotlin/io/github/lishangbu/avalon/oauth2/common/web/authentication/DefaultAuthenticationSuccessHandler.kt
package io.github.lishangbu.avalon.oauth2.common.web.authentication

import io.github.lishangbu.avalon.web.util.JsonResponseWriter
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.security.core.Authentication
import org.springframework.security.web.authentication.AuthenticationSuccessHandler
import tools.jackson.databind.json.JsonMapper

/**
 * 认证成功响应处理器
 *
 * 在认证成功后写回统一的 JSON 成功结果
 */
class DefaultAuthenticationSuccessHandler(
    jsonMapper: JsonMapper,
) : AuthenticationSuccessHandler {
    /** JSON 映射器 */
    private val jsonMapper = jsonMapper

    /** 将认证成功结果写入响应体 */
    override fun onAuthenticationSuccess(
        request: HttpServletRequest,
        response: HttpServletResponse,
        authentication: Authentication,
    ) {
        JsonResponseWriter.writeSuccessResponse(response, jsonMapper)
    }
}
